### PR TITLE
Rfink/upgradle

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

**Comment by uschi2000** [comment]
@markelliot

File: src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+
+    @CompileStatic(TypeCheckingMode.SKIP)
+    protected File createFile(String path, File baseDir = projectDir) {
+        File file = file(path, baseDir)
+        assert !file.exists()
+        directory(file.parent, baseDir)
+        assert file.createNewFile()
+        file
```
Comment (markelliot) [inline src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy]: prefer the explicit `return`

File: src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    protected File file(String path, File baseDir = projectDir) {
+        def splitted = path.split('/')
+        def directory = splitted.size() > 1 ? directory(splitted[0..-2].join('/'), baseDir) : baseDir
+        def file = new File(directory, splitted[-1])
+        file
```
Comment (markelliot) [inline src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy]: `return`

File: src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    protected File directory(String path, File baseDir = projectDir) {
+        new File(baseDir, path).with {
```
Comment (markelliot) [inline src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy]: `return`

File: src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            mkdirs()
+            it
```
Comment (markelliot) [inline src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy]: ...and `return`

**Comment by markelliot** [comment]
lgtm except for the `return` nits

File: src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy (hunk 1 above)
Comment (uschi2000) [inline src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy]: done

File: src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy (hunk 2 above)
Comment (uschi2000) [inline src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy]: done

File: src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy (hunk 3 above)
Comment (uschi2000) [inline src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy]: done

File: src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy (hunk 4 above)
Comment (uschi2000) [inline src/test/groovy/com/palantir/gradle/javadist/GradleTestSpec.groovy]: done

**Comment by uschi2000** [comment]
squash CR responses

**Pull request merged**